src/rgw/rgw_rest_s3.cc

```diff
@@ -3607,6 +3612,7 @@ int RGW_Auth_S3::authorize_v4(RGWRados *store, struct req_state *s, bool force_b
     return -ENOMEM;
   }
 
+  const char *session_token = nullptr;
```

Comment:
@pritha-srivastava I believe would be good to use uint8_t instead of char.
what's your thought?